### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by artalar**

### CODE REVIEW

File: packages/core/src/__tests__/index.ts [hunk 1]
```diff
@@ -54,80 +54,95 @@ describe('@reatom/core', () => {
...
+        )
       })
-    })
-    test('throw error if declareAtom called with undefined initial state', () => {
-      const run = () => declareAtom(['test'], undefined, reduce => [])
+      test('throw error if atom produced undefined value', () => {
+        const action = declareAction()
+
+        expect(() =>
+          declareAtom(['myAtom'], {}, r => r(action, () => undefined as any))(
```
Comment (belozer) [inline packages/core/src/__tests__/index.ts]: The current file uses `reduce`, `r` and `on`. The desired uniformity so as not to drive you crazy readers.

Comment (belozer) [reply packages/core/src/__tests__/index.ts]: I suggest to fix `on`

File: packages/core/src/createStore.ts [hunk 2]
```diff
@@ -28,25 +32,31 @@ export function createStore(atom: Atom<any>, initState?: State): Store
...
+        assign(state, initState)
+      else if (initState !== undefined) throwError('Invalid initial state')
+
+      storeTree.union(getTree(atom as Atom<any>))
+      const ctx = createCtx(state, initAction)
+      storeTree.forEach(initAction.type, ctx)
+      assign(state, ctx.stateNew)
+      initialAtoms = new Set(Object.keys(ctx.stateNew))
+    }
+  }
```
Comment (Wroud) [inline packages/core/src/createStore.ts]: I think there can be more simple than tons of 'if'

File: packages/core/src/declareAtom.ts [hunk 3]
```diff
@@ -121,8 +121,8 @@ export function declareAtom<TState>(
     depTree.fnsMap.forEach((_, key) => _tree.addFn(update, key))
   }
 
-  handler(_initAction, (state = initialState as TState) => state)
-  dependencyMatcher(handler)
+  on(_initAction, (state = initialState as TState) => state)
+  dependencyMatcher(on)
```
Comment (Wroud) [inline packages/core/src/declareAtom.ts]: Maybe keep handler? it is more descriptive

Comment (artalar) [reply packages/core/src/declareAtom.ts]: `on` is friendly for new contributors

**Comment by belozer** [comment]
fix(core): naming in tests

fix - needs use only for fix bugs. In this case needs use one of `style`, `refactor`, or `chore`

File: packages/core/src/createStore.ts (hunk 2 above)
Comment (artalar) [reply packages/core/src/createStore.ts]: let propose a changes in a new PR

**Pull request merged**